airflow/models/__init__.py

```diff
@@ -1014,9 +1019,8 @@ def are_dependents_done(self, session=None):
         count = ti[0][0]
         return count == len(task.downstream_task_ids)
 
-    @property
     @provide_session
-    def previous_ti(self, session=None):
+    def get_previous_ti(self, session=None):
```

Comment:
Why the extra method?

Reply:
`mypy` raises an error when a property has arity >1 (which makes sense) or when a property is decorated (which might make less sense). I got around those errors by factoring those methods into a non-property that uses the `provide_session` decorator and takes a `session` argument and a property that invokes the non-property.

Alternatively, we could ignore the `mypy` warnings and revert these changes, or not use the decorator here and explicitly create a session instead.